src/dune_rules/context.ml

```diff
@@ -302,20 +297,26 @@ module Build_environment_kind = struct
           match opam_prefix with
           | Some s -> Opam2_environment s
           | None -> Unknown)))
-end
 
-let ocamlfind_printconf_path ~env ~ocamlfind ~toolchain =
-  let args =
-    let args = [ "printconf"; "path" ] in
-    match toolchain with
-    | None -> args
-    | Some s -> "-toolchain" :: Context_name.to_string s :: args
-  in
-  let+ l =
-    Memo.of_reproducible_fiber
-      (Process.run_capture_lines ~display:Quiet ~env Strict ocamlfind args)
-  in
-  List.map l ~f:Path.of_filename_relative_to_initial_cwd
+  let findlib_paths ocamlfind ~kind ~context ~findlib_toolchain ~env ~dir =
+    match (ocamlfind, query ~kind ~findlib_toolchain ~env) with
+    | ( Some ocamlfind
+      , ( Cross_compilation_using_findlib_toolchain _
+        | Opam2_environment _
+        | Unknown ) ) -> Ocamlfind.conf_path ocamlfind
```

Comment:
callout: we're also overriding the opam environment findlib.conf lookup if `OCAMLFIND_CONF` is set. This is consistent with the other behavior, and I've verified with @toots that `opam-cross` doesn't rely on `OCAMLFIND_CONF`.